### PR TITLE
Run voting tests against all three voting modules.

### PR DESCRIPTION
**Pull request by ezekiiel**
Also updates dependencies so that we have the latest and greatest
versions.

### CODE REVIEW

**Comment by JakeHartnell** [comment]
Very cool.

File: contracts/cw-proposal-single/src/tests.rs [hunk 1]
```diff
@@ -387,489 +573,198 @@ fn do_test_votes(
...
-            threshold: PercentageThreshold::Majority {},
-            quorum: PercentageThreshold::Majority {},
-        },
-        Status::Rejected,
-        None,
-        None,
-    );
+    testing::test_majority_vs_half(do_votes_cw20_balances);
+    testing::test_majority_vs_half(do_votes_cw4_weights);
+    testing::test_votes_low_threshold(do_votes_staked_balances)
```
Comment (Callum-A) [inline contracts/cw-proposal-single/src/tests.rs:744]: copy paste error? shouldnt this be `test_majority_vs_half` also?

Comment (ezekiiel) [reply contracts/cw-proposal-single/src/tests.rs]: Great catch! Fixed. Thanks.

**Review by Callum-A — APPROVED** [review]

**Comment by ezekiiel** [comment]
Rebased on main.

**Review by janitachalam — APPROVED** [review]

**Pull request merged**